certified-connectors/Cognite Data Fusion/apiProperties.json

```diff
@@ -5,31 +5,177 @@
     ],
     "connectionParameters": {
       "token": {
-        "type": "oauthSetting",
+        "type": "oAuthSetting",
         "oAuthSettings": {
-          "identityProvider": "oauth2",
+          "identityProvider": "oauth2generic",
           "clientId": "PLACEHOLDER_CLIENTID",
-          "scopes": ["https://bluefield.cognitedata.com/.default"],
+          "scopes": ["{Scope}"],
           "redirectMode": "Global",
-          "redirectUrl": "https://global.consent.azure-apim.net/redirect",
+          "redirectUrl": "{RedirectUrl}",
           "properties": {
-            "IsFirstParty": "False"
+            "IsFirstParty": "False",
+            "IsOnbehalfofLoginSupported": true
           },
           "customParameters": {
-            "authorizationUrl": {
-              "value": "https://login.microsoftonline.com/PLACEHOLDER_TENANTID/oauth2/v2.0/authorize"
+            "authorizationUrlTemplate": {
+              "value": "{AuthorizationUrl}"
             },
-            "tokenUrl": {
-              "value": "https://login.microsoftonline.com/PLACEHOLDER_TENANTID/oauth2/v2.0/token"
+            "authorizationUrlQueryStringTemplate":{
+               "value":"client_id={ClientId}&response_type=code&redirect_uri={RedirectUrl}&scope={Scope}&state={State}"
             },
-            "refreshUrl": {
-              "value": "https://login.microsoftonline.com/PLACEHOLDER_TENANTID/oauth2/v2.0/token"
+            "tokenUrlTemplate": {
+              "value": "{TokenUrl}"
+            },
+            "tokenBodyTemplate":{
+               "value":"code={code}&grant_type=authorization_code&redirect_uri={RedirectUrl}&client_id={ClientId}&client_secret={ClientSecret}&scope={Scope}"
+            },
+            "refreshUrlTemplate": {
+              "value": "{RefreshUrl}"
+            },
+            "refreshBodyTemplate":{
+               "value":"code={code}&grant_type=authorization_code&redirect_uri={RedirectUrl}&client_id={ClientId}&client_secret={ClientSecret}&scope={Scope}"
             }
           }
+        },
+        "uiDefinition": {
+          "displayName": "Login",
+          "description": "Sign in with your Cognite Data Fusion credentials",
+          "tooltip": "Sign in with your Cognite Data Fusion credentials",
+          "constraints": {
+            "required": "true"
+          }
+        }
+      },
+      "token:CDFCluster": {
+        "type": "string",
+        "uiDefinition": {
+          "constraints": {
+            "required": "true"
+          },
+          "displayName": "Enter your Cognite Data Fusion Cluster.",
+          "description": "Enter your Cognite Data Fusion Cluster.",
+          "tooltip": "api, greenfield, bluefield, az-eastus-1, etc..."
+        }
+      },
+      "token:TenantId": {
+        "type": "string",
+        "uiDefinition": {
```

Comment:
Please fix below errors. 

##[error]@{level=Critical; message=Missing metadata.; valuePath=$.properties.connectionParameters.token:TenantId.metadata; value=}

##[error]@{level=Critical; message=Missing hidden.; valuePath=$.properties.connectionParameters.token:TenantId.uiDefinition.constraints; value=}

##[error]@{level=Critical; message=Invalid value for the description property within the UI Definition object.; valuePath=$.properties.connectionParameters.token:TenantId.uiDefinition.description; value=Enter your Tenant ID.}

##[error]@{level=Critical; message=Invalid value for the display name property within the UI Defintion object.; valuePath=$.properties.connectionParameters.token:TenantId.uiDefinition.displayName; value=Enter your Tenant ID.}

Please refer the below link for reference. 

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/schemas/paconn-apiProperties.schema.json#L214